testplan/web_ui/testing/src/state/store.js

```diff
@@ -0,0 +1,127 @@
+// eslint-disable-next-line max-len
+const DISABLE_REDUX_DEVTOOLS = process.env.REACT_APP_DISABLE_REDUX_DEVTOOLS === 'true';
+let shaveObjectEntriesFromObject = null;
+if(__DEV__ && FIX_DEVTOOL_SPEED && !DISABLE_REDUX_DEVTOOLS) {
+  console.warn(flattened`
+    You've set the environment variable REACT_APP_FIX_DEVTOOL_SPEED=true
+    which will disable several features in Redux Devtools.
+  `);
+  const _isObjectLike = require('lodash/isObjectLike');
+  const _isFunction = require('lodash/isFunction');
+  shaveObjectEntriesFromObject = (entry, placeholder = '<<omitted>>') => (
+    !_isObjectLike(entry) ? entry : Object.fromEntries(
+      Object.entries(entry).map(([ prop, val ]) => [
+        prop, _isObjectLike(val)
+          ? _isFunction(placeholder) ? placeholder(prop, val) : placeholder
+          : val
+      ])
+  ));
+}
+
+const createReducer = () => combineReducers({
+  app: require('./appSlice').default.reducer,
+  // eslint-disable-next-line max-len
+  report: require('../Report/BatchReportBeta/state/reportSlice').default.reducer,
+  ui: require('../Report/BatchReportBeta/state/uiSlice').default.reducer,
+});
+
+const store = configureStore({
+  reducer: createReducer(),
+  middleware: [
+    ...getDefaultMiddleware({
+      thunk: {
+        extraArgument: { uiHistory, appHistory }
+      },
+      serializableCheck: __DEV__ && (!FIX_DEVTOOL_SPEED ? true : {
+        ignoredPaths: [
+          'report.document',
+          'ui.selectedEntry',
+        ],
+        ignoredActions: [
+          fetchReport.fulfilled.type,
+          setSelectedEntry.fulfilled.type,
+        ],
+      }),
+      immutableCheck: __DEV__ && (!FIX_DEVTOOL_SPEED ? true : {
+        ignoredPaths: [
+          'report.document',
+          'ui.selectedEntry',
+        ],
+      }),
+    })
+  ],
+  devTools: __DEV__ && !DISABLE_REDUX_DEVTOOLS && (!FIX_DEVTOOL_SPEED ? true : {
+    name: 'testplan',
+    maxAge: 10,
+    trace: true,
+    traceLimit: 3,
+    shouldCatchErrors: true,
+    shouldRecordChanges: true,
+    shouldHotReload: true,
+    actionSanitizer: action => {
+      switch(action.type) {
+        case fetchReport.fulfilled.type:
+          return {
+            ...action,
+            payload: '<<omitted>>',
+          };
+        case setSelectedEntry.fulfilled.type:
+          return {
+            ...action,
+            payload: shaveObjectEntriesFromObject(action.payload)
+          };
+        default:
+          return action;
+      }
+    },
+    stateSanitizer: state => {
+      // devtools crashes and the UI freezes at 45_864_416 so this number can
+      // likely be tuned up or down (right now it's just a guess)
+      if(35_000_000 < getLastResponseContentLength(state)) {
+        // these slow down the extension - and hence the UI - a lot
+        window.__CURRENT_REPORT = state.report.document;
+        window.__SELECTED_ENTRY = state.ui.selectedEntry;
```

Comment:
Why are we using window while redux store is supposed to be global accessible?

Reply:
this is just to help debugging.